### PR TITLE
cpu_model_inter_generation: New test case added

### PR DESCRIPTION
**Pull request by vivianQizhu**
Id: 1873908
Signed-off-by: Qianqian Zhu <qizhu@redhat.com>

### CODE REVIEW

**Comment by vivianQizhu** [comment]
@nanliu-r Please help review, thanks.
 (1/2) Host_RHEL.m8.u3.product_rhel.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.cpu_model_inter_generation.q35: PASS (83.45 s)
 (2/2) Host_RHEL.m8.u3.product_rhel.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.cpu_model_inter_generation.enforce.q35: PASS (8.00 s)

File: qemu/tests/cpu_model_inter_generation.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        finally:
+            if vm and vm.is_alive():
+                vm.destroy()
+
+    vm_name = params['main_vm']
+    old_cpu_models = params['old_cpu_model']
+    new_cpu_models = params['new_cpu_model']
+    warning_text = params.get('warning_text')
+    boot_expected = params.get('boot_expected', 'yes')
+    host_cpu_model = domcapability_xml.DomCapabilityXML().get_hostmodel_name()
```
Comment (nanliu-r) [inline qemu/tests/cpu_model_inter_generation.py:53]: Hi, there will be an error if the host has no libvirt installed. Seems we won't install libvirt default.

File: qemu/tests/cpu_model_inter_generation.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                          ' when cpu enforce mode is on')
+        finally:
+            if vm and vm.is_alive():
+                vm.destroy()
+
+    fd = open("/proc/cpuinfo")
+    cpu_info = fd.read()
+    fd.close()
+    vendor = cpu.get_cpu_vendor(cpu_info)
+    latest_cpu_model = cpu.CPU_TYPES.get(vendor)[0]
```
Comment (nanliu-r) [inline qemu/tests/cpu_model_inter_generation.py:52]: Hi, we need to check if the `latest_cpu_model` is supported by current qemu version, if no, we need to pick up the next cpu model in cpu_types.

Comment (vivianQizhu) [reply qemu/tests/cpu_model_inter_generation.py]: @nanliu-r Updated, please help check, thanks.

File: qemu/tests/cpu_model_inter_generation.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                if warning_text not in str(e):
+                    raise
+            else:
+                raise
+        else:
+            if boot_expected == 'no':
+                test.fail('The vm should not boot successfully'
+                          ' when cpu enforce mode is on')
+        finally:
+            if vm and vm.is_alive():
```
Comment (nanliu-r) [inline qemu/tests/cpu_model_inter_generation.py:46]: Please add VM check `vm.verify_kernel_crash()` in here, others look good to me, thanks.

Comment (vivianQizhu) [reply qemu/tests/cpu_model_inter_generation.py:46]: @nanliu-r Updated, thanks.

**Comment by nanliu-r** [comment]
Test on Opteron_G3 with RHEL.7.9 host PASS. 
 (1/2) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.cpu_model_inter_generation.q35: PASS (29.42 s)
 (2/2) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.cpu_model_inter_generation.enforce.q35: PASS (3.99 s)
(1/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.6.10.x86_64.io-github-autotest-qemu.cpu_model_inter_generation: PASS (30.21 s)
 (2/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.6.10.x86_64.io-github-autotest-qemu.cpu_model_inter_generation.enforce: PASS (4.00 s)
 (5/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.cpu_model_inter_generation: PASS (29.72 s)
 (6/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.cpu_model_inter_generation.enforce: PASS (4.15 s)


Test on Icelake-Server with RHEL.8.3.0 host PASS.
 (1/6) Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.6.10.x86_64.io-github-autotest-qemu.cpu_model_inter_generation: PASS (34.05 s)
 (2/6) Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.6.10.x86_64.io-github-autotest-qemu.cpu_model_inter_generation.enforce: PASS (7.92 s)
 (3/6) Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.cpu_model_inter_generation: PASS (33.82 s)
 (4/6) Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.cpu_model_inter_generation.enforce: PASS (8.11 s)
 (5/6) Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.cpu_model_inter_generation: PASS (34.41 s)
 (6/6) Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.cpu_model_inter_generation.enforce: PASS (8.14 s)


ACK, Thanks.

**Pull request merged**